api/src/db/DynamoDataClient.ts

```diff
@@ -72,6 +72,16 @@ export const DynamoDataClient = (
     }
   };
 
+  const findUserByBusinessName = async (businessName: string): Promise<UserData[]> => {
+    const usersWithBusinesses = await userDataClient.queryUsersWithBusinesses();
+    const normalizedBusinessName = businessName.trim().toLowerCase();
+    return usersWithBusinesses.filter((user: UserData) => {
+      return Object.values(user.businesses).some(
+        (business: Business) => business.profileData?.businessName?.toLowerCase() === normalizedBusinessName
```

Comment:
[boulder] How will this work when business data is no longer inside user data?

Reply:
Currently, the workflow relies on business data being inside the user data structure. In the planned migration, we will remove the businesses object from userData and make slight updates to the code.

We will introduce a GSI (`currentBusinessId`), the common field between the users and businesses tables. The updated flow will:
- Query the businessesTable to find businesses by name with fuzzy (matching: this is now implemented in the current state).
- Extract the currentBusinessId from the matched business.
- Query the usersTable using currentBusinessId to find the associated users.
let me know your thoughts @seidior . this also answers your previous questions on what field do we have that ties both user and businesses currently.

Reply:
To sum up what we discussed in our meeting just now:

- The current migration for splitting out users and businesses and keeping them in sync should be modified (or a follow-up migration should be added) so that, in addition to a version field being added to the business record, we also add a field that is called something like "owningUser" that is the UUID of the user data record.
- We then can modify this work to use the secondary index for business names on the businesses table to find the corresponding business record, then use this "owningUser" field in the business record to get the user data and return that.
  - This means we don't have to search through all businesses of all users for the correct user data, we just have to find the fuzzy-matching businesses and return those user data objects. This should _vastly_ speed up the performance of this endpoint.